### PR TITLE
Added related_entities, note_class to Note

### PR DESCRIPTION
**Pull request by gbuisson**
#close https://github.com/advthreat/iroh/pull/7569

Add new fields to the Note Entity as a mean to manage Incident Task COAs progression.

### CODE REVIEW

File: src/ctim/schemas/note.cljc [hunk 1]
```diff
@@ -20,6 +21,8 @@ For example, an analyst may add a Note to a Campaign object created by another o
    (f/entry :entity_id c/Reference)
    (f/entry :content c/Markdown))
   (f/optional-entries
+   (f/entry :note_class v/NoteClass)
+   (f/entry :related_entities (f/seq-of c/Reference))
```
Comment (ereteog) [inline src/ctim/schemas/note.cljc:25]: I would use `note_type` in order to be consistent with `relationship_type` of [relationship schema](https://github.com/threatgrid/ctim/blob/4aa3a14db97715c493a05dec581984b2ae542af5/src/ctim/schemas/relationship.cljc#L18).
I would also use `entity_ids` like `entity_id` of [feedback](https://github.com/threatgrid/ctim/blob/4aa3a14db97715c493a05dec581984b2ae542af5/src/ctim/schemas/feedback.cljc#L17) and `asset_id` of asset-mappings and asset-properties.

Comment (gbuisson) [reply src/ctim/schemas/note.cljc:25]: I can't use NoteTypenas it conflicts with another eponym field in the vocabularies

Comment (gbuisson) [reply src/ctim/schemas/note.cljc:25]: I debates the use of entity_ids but I wanted perhaps to convey the notion of a primary target for the relation

Comment (gbuisson) [reply src/ctim/schemas/note.cljc]: we finally decided to switch to `related_entities` as a single vector of maps including the target type for convenience.

File: src/ctim/schemas/vocabularies.cljc [hunk 2]
```diff
@@ -1509,3 +1509,16 @@
 (def-enum-type PromotionMethod promotion-method
   :open? false
   :gen (cs/gen promotion-method))
+
+(def note-class
```
Comment (ereteog) [inline src/ctim/schemas/vocabularies.cljc:1513]: if we proxify it, we could make it open and restrict values in the proxy

**Review by ereteog — APPROVED** [review]
LGTM

File: src/ctim/schemas/vocabularies.cljc (hunk 2 above)
Comment (gbuisson) [reply src/ctim/schemas/vocabularies.cljc:1513]: that's a good point

Comment (gbuisson) [reply src/ctim/schemas/vocabularies.cljc]: done

**Pull request merged**